src/webui/CMakeLists.txt

```diff
@@ -1,38 +1,38 @@
-add_library(qbt_webui STATIC
-# headers
-api/apicontroller.h
-api/apierror.h
-api/appcontroller.h
-api/authcontroller.h
-api/freediskspacechecker.h
-api/isessionmanager.h
-api/logcontroller.h
-api/rsscontroller.h
-api/searchcontroller.h
-api/synccontroller.h
-api/torrentscontroller.h
-api/transfercontroller.h
-api/serialize/serialize_torrent.h
-webapplication.h
-webui.h
+add_library(qbt_webui
+    # headers
+    api/apicontroller.h
+    api/apierror.h
+    api/appcontroller.h
+    api/authcontroller.h
+    api/freediskspacechecker.h
+    api/isessionmanager.h
+    api/logcontroller.h
+    api/rsscontroller.h
+    api/searchcontroller.h
+    api/synccontroller.h
+    api/torrentscontroller.h
+    api/transfercontroller.h
+    api/serialize/serialize_torrent.h
+    webapplication.h
+    webui.h
 
-# sources
-api/apicontroller.cpp
-api/apierror.cpp
-api/appcontroller.cpp
-api/authcontroller.cpp
-api/freediskspacechecker.cpp
-api/logcontroller.cpp
-api/rsscontroller.cpp
-api/searchcontroller.cpp
-api/synccontroller.cpp
-api/torrentscontroller.cpp
-api/transfercontroller.cpp
-api/serialize/serialize_torrent.cpp
-webapplication.cpp
-webui.cpp
+    # sources
+    api/apicontroller.cpp
+    api/apierror.cpp
+    api/appcontroller.cpp
+    api/authcontroller.cpp
+    api/freediskspacechecker.cpp
+    api/logcontroller.cpp
+    api/rsscontroller.cpp
+    api/searchcontroller.cpp
+    api/synccontroller.cpp
+    api/torrentscontroller.cpp
+    api/transfercontroller.cpp
+    api/serialize/serialize_torrent.cpp
+    webapplication.cpp
+    webui.cpp
 )
 
-qbt_target_sources(qBittorrent PRIVATE www/webui.qrc)
+target_sources(qbt_webui INTERFACE www/webui.qrc)
```

Comment:
In this new implementation, it has to be `INTERFACE` because app needs access to it. If you try building it with `PRIVATE`, accessing the WebUI will just yield a web page saying "Not Found".

Reply:
> because app needs access to it

I see it. I just don't understand why did you change it. Why all web UI related stuff can't be built into corresponding library?
Well, maybe it's just that the splitting into libraries is still something inappropriate...

Reply:
@glassez 

Actually, I just found out that for Qt, only the main application component needs to know about the resource sources, regardless of what component they are used in. So I have corrected the code to the following:

```diff
diff --git a/src/app/CMakeLists.txt b/src/app/CMakeLists.txt
index d01dfaeb4..567a617cb 100644
--- a/src/app/CMakeLists.txt
+++ b/src/app/CMakeLists.txt
@@ -140,6 +140,7 @@ endif()
 
 if (GUI)
     set_target_properties(qbt_app PROPERTIES OUTPUT_NAME qbittorrent)
+    target_sources(qbt_app PRIVATE "${qBittorrent_SOURCE_DIR}/src/gui/about.qrc")
     target_link_libraries(qbt_app PRIVATE qbt_gui)
 else()
     set_target_properties(qbt_app PROPERTIES OUTPUT_NAME qbittorrent-nox)
@@ -147,6 +148,7 @@ endif()
 
 if (WEBUI)
     target_sources(qbt_app PRIVATE
+        "${qBittorrent_SOURCE_DIR}/src/webui/www/webui.qrc"
         ${QBT_WEBUI_QM_FILES}
         ${qBittorrent_BINARY_DIR}/src/webui/www/translations/webui_translations.qrc # yes, it's supposed to be "*_BINARY_DIR"
     )
diff --git a/src/gui/CMakeLists.txt b/src/gui/CMakeLists.txt
index e8595ce2f..b8188f52c 100644
--- a/src/gui/CMakeLists.txt
+++ b/src/gui/CMakeLists.txt
@@ -188,8 +188,6 @@ add_library(qbt_gui STATIC
     updownratiodialog.ui
 )
 
-target_sources(qbt_gui INTERFACE about.qrc)
-
 target_link_libraries(qbt_gui
     PRIVATE
         qbt_base
diff --git a/src/webui/CMakeLists.txt b/src/webui/CMakeLists.txt
index 51cdfe489..a0d14a70f 100644
--- a/src/webui/CMakeLists.txt
+++ b/src/webui/CMakeLists.txt
@@ -33,6 +33,4 @@ add_library(qbt_webui STATIC
     webui.cpp
 )
 
-target_sources(qbt_webui INTERFACE www/webui.qrc)
-
 target_link_libraries(qbt_webui PRIVATE qbt_base)

```

Reply:
>Actually, I just found out that for Qt, only the main application component needs to know about the resource sources, regardless of what component they are used in

It doesn't seem quite logical that linking 
library is not enough to use it.

Reply:
@glassez 

> It doesn't seem quite logical that linking
library is not enough to use it.

I don't know exactly why either, but that's not enough. Probably due to some Qt implementation detail.

Reply:
>I don't know exactly why either, but that's not enough. Probably due to some Qt implementation detail.

Then I would leave your original version to keep it where it comes from:
```cmake
target_sources(qbt_webui INTERFACE www/webui.qrc)
```
so using the library will only require one instruction, as expected:
```cmake
target_link_libraries(qbt_app PRIVATE qbt_webui)
```

Reply:
@glassez ok, fixed.